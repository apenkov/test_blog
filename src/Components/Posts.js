import React, {Component} from 'react';
import Post from "./Post";

export default class Posts extends Component {
  render() {
    const data = this.props.data;
    const postsTemplate = data.map((item, index)=>{
      return (
        <div key={index}>
         <Post data ={item}/>
        </div>
      )
    })
    return(
      <div>
        {postsTemplate}
      </div>
    )
  }
}
