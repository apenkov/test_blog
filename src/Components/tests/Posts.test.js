import React from 'react';
import renderer from 'react-test-renderer';
import Posts from "../Posts";

describe('posts tests', () => {
  it('render posts', () => {
    const posts = renderer
      .create(<Posts data={[
        {
        title:'first post',
        littleText:'little text',
        text:'big text'
      },{
          title:'second post',
          littleText:'little text',
          text:'big text'
        }
      ]}/>)
      .toJSON();
    expect(posts).toMatchSnapshot();
  })
})
