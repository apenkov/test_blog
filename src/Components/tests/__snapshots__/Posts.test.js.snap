// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`posts tests render posts 1`] = `
<div>
  <div>
    <div
      className="post"
      id={0}
    >
      <p
        className="title"
      >
        first post
      </p>
      <p
        className="preview"
        style={
          Object {
            "visibility": "visible",
          }
        }
      >
        little text
      </p>
      <button
        className="post_readMore"
        href="#"
        onClick={[Function]}
        style={
          Object {
            "visibility": "visible",
          }
        }
      >
        Читать
      </button>
      <p
        className="body"
        dangerouslySetInnerHTML={
          Object {
            "__html": "big text",
          }
        }
        style={
          Object {
            "height": 0,
            "transition": "height 2s ease",
            "visibility": "hidden",
          }
        }
      />
      <button
        className="post_readLess"
        onClick={[Function]}
        style={
          Object {
            "visibility": "hidden",
          }
        }
      >
        <a
          href="#0"
        >
          Скрыть
        </a>
      </button>
    </div>
  </div>
  <div>
    <div
      className="post"
      id={1}
    >
      <p
        className="title"
      >
        second post
      </p>
      <p
        className="preview"
        style={
          Object {
            "visibility": "visible",
          }
        }
      >
        little text
      </p>
      <button
        className="post_readMore"
        href="#"
        onClick={[Function]}
        style={
          Object {
            "visibility": "visible",
          }
        }
      >
        Читать
      </button>
      <p
        className="body"
        dangerouslySetInnerHTML={
          Object {
            "__html": "big text",
          }
        }
        style={
          Object {
            "height": 0,
            "transition": "height 2s ease",
            "visibility": "hidden",
          }
        }
      />
      <button
        className="post_readLess"
        onClick={[Function]}
        style={
          Object {
            "visibility": "hidden",
          }
        }
      >
        <a
          href="#1"
        >
          Скрыть
        </a>
      </button>
    </div>
  </div>
</div>
`;
